internal/validators/permissions.go

```diff
@@ -0,0 +1,158 @@
+package validators
+
+import (
+	"fmt"
+	"strings"
+)
+
+const (
+	wildcard = "*"
+)
+
+// allCandidateActionsPermitted returns whether a list of Azure RBAC actions should be permitted
+// based on a list of "actions" and "not actions" for a role definition from the Azure APIs, where
+// actions are actions that the role definition explicitly permits and not actons are actions that
+// the role definition explicitly denies, even if they would have been permitted based on the
+// actions (not actions override actions).
+//
+// Wildcards are take into account, but only for the actions and not actions, with one wildcard
+// permitted in the action or not action. Any more than that are invalid. Also, candidate actions do
+// not support any wildcards. The args will be considered invalid and an error will be returned if
+// these wildcard rules are not followed.
+func allCandidateActionsPermitted(candidateActions, actions, notActions []string) (bool, error) {
```

Comment:
:wrench: Please refactor to return a slice of failure messages to provide users with specific feedback regarding which actions/data actions were not permitted.